packages/lib/CalendarService.ts

```diff
@@ -37,6 +37,22 @@ const DEFAULT_CALENDAR_TYPE = "caldav";
 
 const CALENDSO_ENCRYPTION_KEY = process.env.CALENDSO_ENCRYPTION_KEY || "";
 
+function hasFileExtension(url: string): boolean {
+  // Get the last portion of the URL (after the last '/')
+  const fileName = url.substring(url.lastIndexOf("/") + 1);
+  // Check if the file name has a '.' in it and no '/' after the '.'
+  return fileName.includes(".") && !fileName.substring(fileName.lastIndexOf(".")).includes("/");
```

Comment:
```javascript
!fileName.substring(fileName.lastIndexOf(".")).includes("/");
// ^ This check is impossible, redundant I think
```

Reply:
You're right. This also won't cover URL with query parameters like: `https://example.com/remote.php/dav/calendars/example/?someUrlQuery=/i-will-fail.ics`